### PR TITLE
[Enhance] Support augmented training data and ground truth visualization 

### PR DESCRIPTION
**Pull request by ZCMax**
Thanks for your contribution and we appreciate it a lot. The following instructions would make your pull request more healthy and more easily get feedback. If you do not understand some items, don't worry, just make the pull request and seek help from maintainers.

## Motivation
browse_dataset.py only support visualize original training datasets and gt. This PR supports visualize datasets and gt after training data augmentation for better check the data augmentation correctness.

## Modification
add browse_aug_dataset.py in tools/misc folder. This script should be considered to merged into existing browse_dataset.py for 
better use.

## BC-breaking (Optional)
no

## Use cases (Optional)
N/A

## Checklist

1. Pre-commit or other linting tools are used to fix the potential lint issues.
2. The modification is covered by complete unit tests. If not, please add more unit test to ensure the correctness.
3. If the modification has potential influence on downstream projects, this PR should be tested with downstream projects.
4. The documentation has been modified accordingly, like docstring or example tutorials.

### CODE REVIEW

**Comment by ZCMax** [comment]
This PR stays for discussion. Whether to merge browe_aug_dataset.py into existing browse_dataset.py.

File: tools/misc/browse_aug_dataset.py [hunk 1]
```diff
@@ -0,0 +1,220 @@
...
+    if cfg.data.train['type'] == 'ConcatDataset':
+        cfg.data.train = cfg.data.train.datasets[0]
+    train_data_cfg = cfg.data.train
+
+    # if browse_aug_data:
+    #     show_pipeline = cfg.train_pipeline
+    #     print('browse augment data!')
+    # else:
+    #     show_pipeline = cfg.eval_pipeline
+    #     print('browse original data!')
```
Comment (Tai-Wang) [inline tools/misc/browse_aug_dataset.py:76]: Are these codes commented correctly?

**Comment by ZwwWayne** [comment]
> This PR stays for discussion. Whether to merge browe_aug_dataset.py into existing browse_dataset.py.

I think we should merge and allow an option like --aug

File: tools/misc/browse_aug_dataset.py [hunk 2]
```diff
@@ -0,0 +1,220 @@
...
+        'in xxx=yyy format will be merged into config file. If the value to '
+        'be overwritten is a list, it should be like key="[a,b]" or key=a,b '
+        'It also allows nested list/tuple values, e.g. key="[(a,b),(c,d)]" '
+        'Note that the quotation marks are necessary and that no white space '
+        'is allowed.')
+    args = parser.parse_args()
+    return args
+
+
+def build_data_cfg(config_path, skip_type, browse_aug_data, cfg_options):
```
Comment (wHao-Wu) [inline tools/misc/browse_aug_dataset.py:55]: The `browse_aug_data` has not been utilized in this function.

File: tools/misc/browse_aug_dataset.py [hunk 3]
```diff
@@ -0,0 +1,220 @@
...
+            img, None, None, None, out_dir, filename, show=show)
+
+
+def main():
+    args = parse_args()
+
+    if args.output_dir is not None:
+        mkdir_or_exist(args.output_dir)
+
+    cfg = build_data_cfg(args.config, args.skip_type, args.cfg_options)
```
Comment (wHao-Wu) [inline tools/misc/browse_aug_dataset.py:190]: Also, the `browse_aug_data` is a required item if not default value is set.

**Comment by ZCMax** [comment]
> 

If we want to merge these two scripts, we may change the eval_pipeline in all config files.

**Comment by ZCMax** [comment]
Support original dataset visualization and augmented dataset visualization at the same time in browse_dataset.py by adding a option '--aug'.

File: docs/useful_tools.md [hunk 4]
```diff
@@ -96,6 +96,12 @@ python tools/misc/browse_dataset.py configs/_base_/datasets/kitti-3d-3class.py -
 
 **Notice**: Once specifying `--output-dir`, the images of views specified by users will be saved when pressing `_ESC_` in open3d window. If you don't have a monitor, you can remove the `--online` flag to only save the visualization results and browse them offline.
 
+Besides, in order to facilitate the verification of the consistency of the data and the ground-truth after data augmentation, and to view the effect of data augmentation, we also support visualize the augmented dataset by simply adding the `--aug` flag.
```
Comment (Tai-Wang) [inline docs/useful_tools.md:99]: support visualization of augmented input data by xxx

File: tools/misc/browse_dataset.py [hunk 5]
```diff
@@ -67,10 +71,17 @@ def build_data_cfg(config_path, skip_type, cfg_options):
...
-    # eval_pipeline purely consists of loading functions
-    # use eval_pipeline for data loading
+
+    if aug:
+        show_pipeline = cfg.train_pipeline
+    else:
+        show_pipeline = cfg.eval_pipeline
+        for i in range(len(cfg.train_pipeline)):
+            if cfg.train_pipeline[i].pop('type') == 'LoadAnnotations3D':
+                show_pipeline.insert(1, cfg.train_pipeline[i])
```
Comment (Tai-Wang) [inline tools/misc/browse_dataset.py:81]: Is this "1" a hard code?

File: tools/misc/browse_dataset.py [hunk 6]
```diff
@@ -87,13 +98,14 @@ def to_depth_mode(points, bboxes):
     return points, bboxes
 
 
-def show_det_data(idx, dataset, out_dir, filename, show=False):
+def show_det_data(item, out_dir, show=False):
```
Comment (Tai-Wang) [inline tools/misc/browse_dataset.py:101]: item -> input/data

File: tools/misc/browse_dataset.py [hunk 7]
```diff
@@ -187,53 +192,34 @@ def main():
...
-    cfg = build_data_cfg(args.config, args.skip_type, args.cfg_options)
+    cfg = build_data_cfg(args.config, args.skip_type, args.aug,
+                         args.cfg_options)
     try:
         dataset = build_dataset(
             cfg.data.train, default_args=dict(filter_empty_gt=False))
     except TypeError:  # seg dataset doesn't have `filter_empty_gt` key
         dataset = build_dataset(cfg.data.train)
-    data_infos = dataset.data_infos
+    # data_infos = dataset.data_infos
```
Comment (Tai-Wang) [inline tools/misc/browse_dataset.py:202]: Can remove?

File: docs/useful_tools.md [hunk 8]
```diff
@@ -96,6 +96,12 @@ python tools/misc/browse_dataset.py configs/_base_/datasets/kitti-3d-3class.py -
 
 **Notice**: Once specifying `--output-dir`, the images of views specified by users will be saved when pressing `_ESC_` in open3d window. If you don't have a monitor, you can remove the `--online` flag to only save the visualization results and browse them offline.
 
+Besides, in order to facilitate the verification of the consistency of the data and the ground-truth after data augmentation, and to view the effect of data augmentation, we also support the visualization of augmented input data by simply adding the `--aug` flag.
```
Comment (ZwwWayne) [inline docs/useful_tools.md:99]: To verify the data consistency and the effect of data augmentation, you can also add `--aug` flag to visualize the data after data augmentation using the command as below:

**Review by ZwwWayne — APPROVED** [review]

**Comment by ZwwWayne** [comment]
Need to resolve conflicts of documentation.

**Pull request merged**